File: 프로그래머스0(61~90번)/76.옹알이 (1).js
// 레벨 0
// https://school.programmers.co.kr/learn/courses/30/lessons/120956

/*
머쓱이의 조카가 발음할 수 있는 단어의 개수를 return하도록 solution 함수를 완성해주세요.
*/

function solution(babbling) {
    var answer = 0;
    let can = ["aya", "ye", "woo", "ma"];
  
    for (let i in babbling) {
      let init = babbling[i];
  
      for (let j in can) {
        if (babbling[i].includes(can[j])) {
          init = init.replace(can[j], "X");  // 할 수 있는 단어는 X로 치환
        }
      }
  
      init = init.replace(/X/gi, "");  // X를 모두 공백으로 치환하고 나서
      if (init.length === 0) {  // 공백이 되면 answer에 추가
        answer += 1;
      }
    }
    return answer;
  }

File: 프로그래머스 코테(java)/프로그래머스0(61~90번)/76.옹알이 (1).java
// 레벨 0
// https://school.programmers.co.kr/learn/courses/30/lessons/120956

class Solution {
  public int solution(String[] babbling) {
      String[] str = {"aya", "ye", "woo", "ma"};
      int answer = 0;
      
      for(int i=0; i<babbling.length; i++){
          for(int j=0; j<str.length; j++){
              babbling[i] = babbling[i].replace(str[j], " ");
          }
          
          if(babbling[i].trim().length() == 0){
              answer++;
          }
      }
      return answer;
  }
}